docs/subnets/customize-a-subnet.md

```diff
@@ -211,69 +211,100 @@ If `allowFeeRecipients` feature is enabled on the Subnet, but a validator doesn'
+  "contractDeployerAllowListConfig": {
+    "blockTimestamp": 0,
+    "adminAddresses": ["0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"]
+  }
 }
 ```
 
-If you attempt to add a `Deployer` and you are not an `Admin`, you will see
+In this example, `0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC` is named as the
+`Admin` of the `ContractDeployerAllowList`. This enables it to add other `Admin` or to add
+`Enabled` addresses. Both `Admin` and `Enabled` can deploy contracts. To provide
+a great UX with factory contracts, the `tx.Origin` is checked for being a valid
+deployer instead of the caller of `CREATE`. This means that factory contracts will still be
+able to create new contracts as long as the sender of the original transaction is an allow
+listed deployer.
+
+The `Stateful Precompile` contract powering the `ContractDeployerAllowList` adheres to the [AllowList Solidity interface](#allowlist-interface) at `0x0200000000000000000000000000000000000000` (you can load this interface and interact directly in Remix):
+
+If you attempt to add a `Enabled` and you are not an `Admin`, you will see
 something like:
 ![admin fail](/img/admin_fail.png)
 
 If you attempt to deploy a contract but you are not an `Admin` not
-a `Deployer`, you will see something like:
+a `Enabled`, you will see something like:
 ![deploy fail](/img/deploy_fail.png)
 
-The allow list has three roles: `None`, `Deployer`, and `Admin`.
-
-If you call `readAllowList(addr)` then you can read the current role of `addr`, which will return a uint256 with a value of 0, 1, or 2, corresponding to the roles `None`, `Deployer`, and `Admin` respectively.
+If you call `readAllowList(addr)` then you can read the current role of `addr`, which will return a uint256 with a value of 0, 1, or 2, corresponding to the roles `None`, `Enabled`, and `Admin` respectively.
 
 WARNING: if you remove all of the admins from the allow list, it will no longer be possible to update the allow list without modifying the subnet-evm to schedule a network upgrade.
 
+#### Initial Configuration
+
+It's possible to enable this precompile with an initial configuration to activate it's effect on activation timestamp. This provides a way to enable the precompile without an admin address to manage the deployer list. With this, you can define a list of addresses that are allowed to deploy contracts. Since there will be no admin address to manage the deployer list, it can only be modified through a network upgrade. To use initial configuration, you need to specify addresses in `enabledAddresses` field in your genesis or upgrade file:
+
+```json
+{
+  "contractDeployerAllowListConfig": {
+    "blockTimestamp": 0,
+    "enabledAddresses": ["0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC"]
+  }
+}
+```
+
+This will allow only `0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC` to deploy contracts. For further information about precompile initial configurations see [Initial Configs](#initial-configs).
+
 ### Restricting Who Can Submit Transactions
```

Comment:
```suggestion
#### Restricting Who Can Submit Transactions
```

Reply:
all precompiles use allowlist. so grouping only these two with allowlist might be confusing.